runtime/v1/shim/service.go

```diff
@@ -553,12 +553,12 @@ func shouldKillAllOnExit(bundlePath string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	json.Unmarshal(bundleConfigContents, &bundleSpec)
-
-	if bundleSpec.Linux != nil {
-		for _, ns := range bundleSpec.Linux.Namespaces {
-			if ns.Type == specs.PIDNamespace && ns.Path == "" {
-				return false, nil
+	if err := json.Unmarshal(bundleConfigContents, &bundleSpec); err == nil {
```

Comment:
What about the case if `err != nil` here?

Reply:
Since the content of bundleSpec is not deterministic, we shouldn't interpret bundleSpec.Linux when err != nil

Reply:
Not even to log it somewhere?

Reply:
just return err on non-nil err

Reply:
Origin design is that the reaper will handle any child processes created by shim, including runc. bundleSpec might not be created or be deleted. IMO, no need to log or return error